src/include/btree_inline.h

```diff
@@ -2336,36 +2338,54 @@ __wt_btcur_skip_page(
      */
     if (previous_state == WT_REF_DELETED && __wt_page_del_visible(session, ref->page_del, true)) {
         *skipp = true;
+        WT_STAT_CONN_DATA_INCR(session, cursor_tree_walk_del_page_skip);
         goto unlock;
     }
 
-    /*
-     * Look at the disk address, if it exists, and if the page is unmodified. We must skip this test
-     * if the page has been modified since it was reconciled, since neither the delete information
-     * nor the timestamp information is necessarily up to date.
-     */
-    if ((previous_state == WT_REF_DISK ||
-          (previous_state == WT_REF_MEM && !__wt_page_is_modified(ref->page))) &&
-      __wt_ref_addr_copy(session, ref, &addr)) {
+    if (previous_state == WT_REF_MEM && !__wt_page_is_modified(ref->page)) {
+        /*
+         * Lock the page to prevent concurrent checkpoint reconciliation on the same page while we
```

Comment:
Let's talk more in person about two cases:
1. reconciliation racing with this code
2. racing with serial insert trying to add new keys and marking the page modified

Reply:
Yes, these are the two cases where this change can have impact. I updated the comment to explain the same.